numpy import failure in _cblas_caxpy

When building pyodide locally on Linux Gentoo (system gcc 6.4.0-r1), numpy fails to load, on firefox (see full log below),

<details>

```
________________________________________________________________ test_import[firefox-numpy] _________________________________________________________________

name = 'numpy', selenium = <test.conftest.FirefoxWrapper object at 0x7f1496993630>

    @pytest.mark.parametrize('name', registered_packages())
    def test_import(name, selenium):
        [...]
>           selenium.run('import %s' % import_name)

test/test_common.py:47: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
test/conftest.py:53: in run
    'return pyodide.runPython({!r})'.format(code))
test/conftest.py:60: in run_js
    return self.driver.execute_script(catch)
../../.miniconda3/envs/pyodide-env/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py:635: in execute_script
    'args': converted_args})['value']
../../.miniconda3/envs/pyodide-env/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py:320: in execute
    self.error_handler.check_response(response)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x7f1496993b38>
response = {'status': 500, 'value': '{"value":{"error":"javascript error","message":"Error: Traceback (most recent call last):\\n...1:41937/pyodide.asm.js:4:40671\\n@http://127.0.0.1:41937/test.html:3:26\\n@http://127.0.0.1:41937/test.html:1:2\\n"}}'}

    def check_response(self, response):
        """
        [..]
            raise exception_class(message, screen, stacktrace, alert_text)
>       raise exception_class(message, screen, stacktrace)
E       selenium.common.exceptions.JavascriptException: Message: Error: Traceback (most recent call last):
E         File "/lib/python3.6/site-packages/numpy/core/__init__.py", line 16, in <module>
E           from . import multiarray
E       ImportError: Could not evaluate dynamic lib: /lib/python3.6/site-packages/numpy/core/multiarray.so
E       LinkError: import object field '_cblas_caxpy' is not a Function
E       
E       During handling of the above exception, another exception occurred:
E       
E       Traceback (most recent call last):
E         File "/lib/python3.6/site-packages/pyodide.py", line 33, in eval_code
E           exec(compile(mod, '<exec>', mode='exec'), ns, ns)
E         File "<exec>", line 1, in <module>
E         File "/lib/python3.6/site-packages/numpy/__init__.py", line 142, in <module>
E           from . import add_newdocs
E         File "/lib/python3.6/site-packages/numpy/add_newdocs.py", line 13, in <module>
E           from numpy.lib import add_newdoc
E         File "/lib/python3.6/site-packages/numpy/lib/__init__.py", line 8, in <module>
E           from .type_check import *
E         File "/lib/python3.6/site-packages/numpy/lib/type_check.py", line 11, in <module>
E           import numpy.core.numeric as _nx
E         File "/lib/python3.6/site-packages/numpy/core/__init__.py", line 26, in <module>
E           raise ImportError(msg)
E       ImportError: 
E       Importing the multiarray numpy extension module failed.  Most
E       likely you are trying to import a failed build of numpy.
E       If you're working with a numpy git repo, try `git clean -xdf` (removes all
E       files not under version control).  Otherwise reinstall numpy.
E       
E       Original error was: Could not evaluate dynamic lib: /lib/python3.6/site-packages/numpy/core/multiarray.so
E       LinkError: import object field '_cblas_caxpy' is not a Function

../../.miniconda3/envs/pyodide-env/lib/python3.6/site-packages/selenium/webdriver/remote/errorhandler.py:242: JavascriptException
----------------------------------------------------------------- Captured stdout teardown ------------------------------------------------------------------
Skipping wasm decoding
Python initialization complete
Python exception:
Traceback (most recent call last):
  File "/lib/python3.6/site-packages/numpy/core/__init__.py", line 16, in <module>
    from . import multiarray
ImportError: Could not evaluate dynamic lib: /lib/python3.6/site-packages/numpy/core/multiarray.so
LinkError: import object field '_cblas_caxpy' is not a Function

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/lib/python3.6/site-packages/pyodide.py", line 33, in eval_code
    exec(compile(mod, '<exec>', mode='exec'), ns, ns)
  File "<exec>", line 1, in <module>
  File "/lib/python3.6/site-packages/numpy/__init__.py", line 142, in <module>
    from . import add_newdocs
  File "/lib/python3.6/site-packages/numpy/add_newdocs.py", line 13, in <module>
    from numpy.lib import add_newdoc
  File "/lib/python3.6/site-packages/numpy/lib/__init__.py", line 8, in <module>
    from .type_check import *
  File "/lib/python3.6/site-packages/numpy/lib/type_check.py", line 11, in <module>
    import numpy.core.numeric as _nx
  File "/lib/python3.6/site-packages/numpy/core/__init__.py", line 26, in <module>
    raise ImportError(msg)
ImportError: 
Importing the multiarray numpy extension module failed.  Most
likely you are trying to import a failed build of numpy.
If you're working with a numpy git repo, try `git clean -xdf` (removes all
files not under version control).  Otherwise reinstall numpy.

Original error was: Could not evaluate dynamic lib: /lib/python3.6/site-packages/numpy/core/multiarray.so
LinkError: import object field '_cblas_caxpy' is not a Function


_hiwire_throw_error@http://127.0.0.1:41937/pyodide.asm.js:4:37229
wasm-function[354]@http://127.0.0.1:41937/pyodide_dev.js:813085:1
wasm-function[358]@http://127.0.0.1:41937/pyodide_dev.js:814675:1
pyodide/Module.__runPython@http://127.0.0.1:41937/pyodide.asm.js:4:1877956
_runpython_init_js/Module.runPython@http://127.0.0.1:41937/pyodide.asm.js:4:40671
@http://127.0.0.1:41937/test.html:3:26
@http://127.0.0.1:41937/test.html:1:2
```
</details>

and Chrome,

```
_________________________________________________________________ test_import[chrome-numpy] _________________________________________________________________

name = 'numpy', selenium = <test.conftest.ChromeWrapper object at 0x7f149694c9e8>

    @pytest.mark.parametrize('name', registered_packages())
    def test_import(name, selenium):
[..]
>           selenium.load_package(import_name)

test/test_common.py:46: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
test/conftest.py:67: in load_package
    self.wait.until(PackageLoaded())
[..]
>       raise TimeoutException(message, screen, stacktrace)
E       selenium.common.exceptions.TimeoutException: Message:

../../.miniconda3/envs/pyodide-env/lib/python3.6/site-packages/selenium/webdriver/support/wait.py:80: TimeoutException
----------------------------------------------------------------- Captured stdout teardown ------------------------------------------------------------------
Python initialization complete
Couldn't instantiate wasm: /lib/python3.6/site-packages/numpy/core/multiarray.so 'LinkError: WebAssembly Instantiation: Import #244 module="env" function="_cblas_caxpy" error: function import requires a callable'
Couldn't instantiate wasm: /lib/python3.6/site-packages/numpy/core/umath.so 'ReferenceError: err is not defined'
Couldn't instantiate wasm: /lib/python3.6/site-packages/numpy/core/umath_tests.so 'ReferenceError: err is not defined'
Couldn't instantiate wasm: /lib/python3.6/site-packages/numpy/core/test_rational.so 'ReferenceError: err is not defined'
Couldn't instantiate wasm: /lib/python3.6/site-packages/numpy/core/struct_ufunc_test.so 'ReferenceError: err is not defined'
Couldn't instantiate wasm: /lib/python3.6/site-packages/numpy/core/multiarray_tests.so 'ReferenceError: err is not defined'
Couldn't instantiate wasm: /lib/python3.6/site-packages/numpy/core/operand_flag_tests.so 'ReferenceError: err is not defined'
Couldn't instantiate wasm: /lib/python3.6/site-packages/numpy/fft/fftpack_lite.so 'ReferenceError: err is not defined'
Couldn't instantiate wasm: /lib/python3.6/site-packages/numpy/linalg/lapack_lite.so 'ReferenceError: err is not defined'
Couldn't instantiate wasm: /lib/python3.6/site-packages/numpy/linalg/_umath_linalg.so 'ReferenceError: err is not defined'
Couldn't instantiate wasm: /lib/python3.6/site-packages/numpy/random/mtrand.so 'ReferenceError: err is not defined'
```

Both seemingly due to an issue with `_cblas_caxpy`.  This uses https://github.com/iodide-project/pyodide/pull/98 but it could be reproduced just  by loading and importing numpy.

Not sure what other information might be useful for debugging.

Apart from that, all core cpython tests that pass on Circle CI also pass locally for me.

--- a/packages/numpy/meta.yaml
+++ b/packages/numpy/meta.yaml
@@ -17,6 +17,7 @@ source:
     - patches/init-alloc-cache.patch
     - patches/pass-along-errors-from-init.patch
     - patches/use-dummy-threading.patch
+    - patches/use-local-blas-lapack.patch
 
 build:
   cflags: -include math.h -I../../config

--- a/packages/numpy/patches/use-local-blas-lapack.patch
+++ b/packages/numpy/patches/use-local-blas-lapack.patch
@@ -1,0 +1,17 @@
+diff --git a/numpy/distutils/system_info.py b/numpy/distutils/system_info.py
+old mode 100644
+new mode 100755
+index bea120cf9..5b2ccb2c1
+--- a/numpy/distutils/system_info.py
++++ b/numpy/distutils/system_info.py
+@@ -406,8 +406,8 @@ def get_info(name, notfound_action=0):
+           'numeric': Numeric_info,
+           'numarray': numarray_info,
+           'numerix': numerix_info,
+-          'lapack_opt': lapack_opt_info,
+-          'blas_opt': blas_opt_info,
++          'lapack_opt': lapack_src_info,
++          'blas_opt': blas_src_info,
+           'boost_python': boost_python_info,
+           'agg2': agg2_info,
+           'wx': wx_info,

